src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs

```diff
@@ -324,9 +343,189 @@ public Mapper(WordEmbeddingsTransform parent, Schema inputSchema)
+                var wordVectors = _parent._currentVocab.WordVectors;
+                var tensorD = new List<float>();
+                tensorD.AddRange(wordVectors);
+                tensorD.AddRange(Enumerable.Repeat(0.0f, _parent._currentVocab.Dimension));
+                tensorD.AddRange(Enumerable.Repeat(float.MaxValue, _parent._currentVocab.Dimension));
+                tensorD.AddRange(Enumerable.Repeat(float.MinValue, _parent._currentVocab.Dimension));
+                var nameD = ctx.AddInitializer(tensorD, shapeD, "WordEmbeddingWeights");
+
+                // Allocate F, a value representing an out-of-dictionary word
+                var tensorF = _parent._currentVocab.GetNumWords();
+                var nameF = ctx.AddInitializer(tensorF, "NotFoundValueComp");
+
+                // Retrieve X, name of input
+                var nameX = srcVariableName;
+
+                // Do label encoding
+                var nameY = ctx.AddIntermediateVariable(null, "LabelEncodedInput", true);
+                var nodeY = ctx.CreateNode("LabelEncoder", nameX, nameY, ctx.GetNodeName("LabelEncoder"));
+                nodeY.AddAttribute("classes_strings", _parent._currentVocab.GetWordLabels());
+                nodeY.AddAttribute("default_int64", _parent._currentVocab.GetNumWords());
+
+                // Do steps necessary for min and max embedding vectors
+
+                // Map to boolean vector representing missing words
+                var nameA = ctx.AddIntermediateVariable(null, "NotFoundValuesBool", true);
+                var nodeA = ctx.CreateNode("Equal", new[] { nameY, nameF }, new[] { nameA }, ctx.GetNodeName("Equal"), "");
+
+                // Cast the not found vector to a vector of floats
+                var nameB = ctx.AddIntermediateVariable(null, "NotFoundValuesFloat", true);
+                var nodeB = ctx.CreateNode("Cast", nameA, nameB, ctx.GetNodeName("Cast"), "");
+                nodeB.AddAttribute("to", 1);
+
+                // Scale the not found vector to get to location for max weights
+                var nameSMax = ctx.AddIntermediateVariable(null, "ScaleMax", true);
+                var nodeSMax = ctx.CreateNode("Scale", nameB, nameSMax, ctx.GetNodeName("Scale"), "");
+                nodeSMax.AddAttribute("scale", 2.0);
+
+                // Cast scaled word label locations to ints
+                var nameVMin = ctx.AddIntermediateVariable(null, "CastMin", true);
+                var nodeVMin = ctx.CreateNode("Cast", nameA, nameVMin, ctx.GetNodeName("Cast"), "");
+                nodeVMin.AddAttribute("to", 7);
+
+                var nameVMax = ctx.AddIntermediateVariable(null, "CastMax", true);
+                var nodeVMax = ctx.CreateNode("Cast", nameSMax, nameVMax, ctx.GetNodeName("Cast"), "");
+                nodeVMax.AddAttribute("to", 7);
+
+                // Add the scaled options back to originals
```

Comment:
```suggestion
                // Add the scaled options back to originals. The outputs of the following three Add are almost identical the output of the previous LabelEncoder. The only difference is that out-of-vocab tokens are mapped to k+1 for applying ReduceMin and k+2 for applying ReduceMax so that out-of-vocab tokens do not affect embedding results at all.
```